Switch to Standard Term Support (STS) releases (#7517)
dotnet/core: release-policies.md

# Release Policies

The .NET team uses the following policies for [.NET releases](releases.md).

## Release cadence

New major .NET versions are released annually in November, at [.NET Conf](https://www.dotnetconf.net/).

> Patch updates are released monthly on the second tuesday of each month, also known as Patch Tuesday.

Reviewer: ```suggestion
Patch updates are released monthly on the second Tuesday of each month, also known as Patch Tuesday.
```

> Minor versions of the .NET SDK are releases approximately quarterly. These are known as [feature bands](https://docs.microsoft.com/en-us/dotnet/core/releases-and-support#feature-bands-sdk-only).

Reviewer 2: Suggestion: `are releases approximately quarterly` sounds strange to me. Maybe `release approximately quartlery` would be better.

Reviewer 3: ```suggestion
Minor versions of the .NET SDK are released approximately quarterly. These are known as [feature bands](https://docs.microsoft.com/dotnet/core/releases-and-support#feature-bands-sdk-only).
```

## Release types

> Each .NET release is defined as either **Standard Support** or **Long Term Support (LTS)**, at the beginning of the release.

Author: ```suggestion
Each .NET release is defined as either **Standard Term Support (STS)** or **Long Term Support (LTS)**, at the beginning of the release.
```

> Each .NET release is defined as either  **Standard Support** or **Long Term Support (LTS)**, at the beginning of the release.

Reviewer 3: ```suggestion
Each .NET release is defined as either  **standard support** or **long term support (LTS)**, at the beginning of the release.
```

The release types:

> * **Standard Support** releases are supported for eighteen months, released in even-numbered years. They are intended for users that want to take advantage of the newest features and improvements and to stay on the leading edge of .NET innovation.
> * **LTS** releases are supported for three years, released in odd-numbered years. They are intended for users that want the stability and lower cost of maintaining an application for an extended period, only needing to upgrade their .NET version for security patches.

Author: ```suggestion
* **STS** releases are supported for eighteen months, released in even-numbered years. They are intended for users that want to take advantage of the newest features and improvements and to stay on the leading edge of .NET innovation.
* **LTS** releases are supported for three years, released in odd-numbered years. They are intended for users that want the stability and lower cost of maintaining an application for an extended period, only needing to upgrade their .NET version for security patches.
```

> Note: **Standard** releases were previously called **Current**.

Author: ```suggestion
Note: **Standard Term Support** releases were previously called **Current**.
```

> LTS and Standard Support releases differ only by support duration. The .NET team follows the same software engineering and release processes for both release types, including for security, compatibility, and reliability. Both releases may contain major new features and breaking changes. The .NET team aspires to enable straightforward migration from one release to another, independent of release type.

Author: ```suggestion
LTS and STS releases differ only by support duration. The .NET team follows the same software engineering and release processes for both release types, including for security, compatibility, and reliability. Both releases may contain major new features and breaking changes. The .NET team aspires to enable straightforward migration from one release to another, independent of release type.
```

## Support phases

.NET releases go through multiple support phases, with varying support levels.

* **Preview** releases are not supported but are offered for public testing and for the opportunity to give feedback.

> * **Go-live** support enables users to deploy a pre-release build in production, and are supported by Microsoft. These are typically Release Candidate (RC) releases.

Reviewer 3: ```suggestion
* **Go-live** support enables users to deploy a pre-release build in production, and are supported by Microsoft. These are typically release candidate (RC) releases.
```

> * **Release Candidate** releases have a go-live support that enables users to deploy a pre-release build in production and are supported by Microsoft.

Reviewer 4: "Release Candidate" should be changed to "Go-Live"

> * **Release Candidate** releases have a go-live support that enables users to deploy a pre-release build in production and are supported by Microsoft.

Author: ```suggestion
* **Go-Live** releases are supported by Microsoft in production. These are typically our release candidate builds just before the generally available release.
```

> * **Active** support is provided for the majority of the period after a release is Generally Available (GA). Functional and security improvements will be provided, including support for new operating system versions.

Reviewer 3: ```suggestion
* **Active** support is provided for the majority of the period after a release is generally available (GA). Functional and security improvements will be provided, including support for new operating system versions.
```

* **Maintenance** support is provided for the last six months of support. Improvements are limited to security fixes. Support for new operating system versions will be provided on a best-effort basis.

> * **End of support** or end-of-life (EOL) marks the point where a release is no longer supported.

Reviewer 5: ```suggestion
* **End of support** or end-of-life (EOL) marks the point where a release no longer receives servicing updates and is no longer supported.
```

Author: I don't see that this addition is necessary. It is fully implied and called out in full detail in a later section.

> * **End of support** or end-of-life (EOL) marks the point where a release is no longer supported.

Reviewer 4: "End of support" should be "End of life"

[Support for various operating systems](os-lifecycle-policy.md) is defined for each release.

## Servicing

Improvements are released as full re-releases of the product, called "patch releases". Patch releases are cumulative. Patches are released on the Microsoft "Patch Tuesday" (second Tuesday of each month), however there is no guarantee that there will be a .NET release on any given Patch Tuesday.

Patches are announced in [release notes](release-notes/README.md), on the [.NET blog](https://devblogs.microsoft.com/dotnet/category/maintenance-and-updates/), and [dotnet/announcements](https://github.com/dotnet/announcements/labels/Monthly-Update).

> Breaking changes are not accepted during servicing, except (in the very rare case) to resolve critical issues, such as a security vulnerability.

Reviewer 5: ```suggestion
Breaking changes are not accepted during servicing, except (in the very rare case) to resolve critical issues, such as a security vulnerability. Breaking changes are documented.
```

## End of support

> As the end of support nears for a given .NET version, we recommend that you move to a newer .NET version, and reduce/remove your use of the given .NET version. After support ends, we recommend that you uninstall a given .NET version if you are no longer using it, or install the latest patch, and accelerate your plans to remove your use of that .NET version.

Author: What wording would you suggest instead?

> As the end of support nears for a given .NET version, we recommend that you move to a newer .NET version, and reduce/remove your use of the given .NET version. After support ends, we recommend that you uninstall a given .NET version if you are no longer using it, or install the latest patch, and accelerate your plans to remove your use of that .NET version.

Reviewer 4: Term should remain "end of life"

> As the end of support nears for a given .NET version, we recommend that you move to a newer .NET version, and reduce/remove your use of the given .NET version. After support ends, we recommend that you uninstall a given .NET version if you are no longer using it, or install the latest patch, and accelerate your plans to remove your use of that .NET version.

Reviewer 5: This isn't about EOL v/s EOS.

> As the end of support nears for a given .NET version, we recommend that you move to a newer .NET version, and reduce/remove your use of the given .NET version. After support ends, we recommend that you uninstall a given .NET version if you are no longer using it, or install the latest patch, and accelerate your plans to remove your use of that .NET version.

Reviewer 5: ```suggestion
As the end of support nears for a given .NET version, we strongly recommend you move to a newer, supported version of .NET. .NET releases that have reached end of support do not get security or other patches, so continuing to use such a version may expose you to unpatched security vulnerabilities.
```
